### PR TITLE
Wrap redis connection to retry on fail.

### PR DESCRIPTION
**Pull request by jmcarp**
So that we can retry connecting to redis if the initial connect fails. Note--we could use a werkzeug `localproxy` instead of a wrapper class, but I often regret doing that.

### CODE REVIEW

File: uaaextras/webapp.py [hunk 1]
```diff
@@ -58,13 +58,24 @@
...
-    logging.warn('Unable to connect to Redis: {0}'.format(exc))
-    r = None
+
+class RedisWrapper(object):
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    @property
+    def redis(self):
+        if not hasattr(self, '_redis'):
```
Comment (LinuxBozo) [inline uaaextras/webapp.py]: How does this get unset as an attr?

Comment (jmcarp) [inline uaaextras/webapp.py]: It's not set in the constructor, so until this method successfully connects to redis, `hasattr(self, '_redis')` should be `False`.

Comment (LinuxBozo) [inline uaaextras/webapp.py]: Right, but I thought the point was that this was trying to solve an issue where there was a lost connection? I could have connected once in the past, still have `_redis` as an attr, but not actually have a valid connection. Maybe I'm misunderstanding the problem?

**Comment by cnelson** [comment]
Bump on this PR;  This caused an issue in production down when k8s redis went away and came back

**Pull request closed without merging**